File: 20221010_Monday/20221010.rb
# Leetcode: 1328. Break a Palindrome.
# https://leetcode.com/problems/break-a-palindrome/
# @param {String} palindrome
# @return {String}
def break_palindrome(palindrome)
  pal = palindrome
  if pal.chars.uniq.join == ?a
    if 1 == pal.size
      return ""
    else
      return pal[0...-1] + ?b
    end
  end
  ans = pal.chars
  (0...ans.size).each do |i|
    char = ans[i]
    next if ?a == char
    ans[i] = (ans[i].ord - 1).chr
    break
  end
  ans.join
end

File: 20221010_Monday/20221010.go
// Leetcode: 1328. Break a Palindrome.
// https://leetcode.com/problems/break-a-palindrome/
// = = = = = = = = = = = = = =
// Accepted.
// Thanks God, Jesus Christ!
// = = = = = = = = = = = = = =
// Runtime: 3 ms, faster than 21.15% of Go online submissions for Break a Palindrome.
// Memory Usage: 2.1 MB, less than 30.77% of Go online submissions for Break a Palindrome.
// 2022.10.10 Daily Challenge.
func breakPalindrome(palindrome string) string {
  pal := []rune(palindrome)
  if 1 == len(pal) {
    return ""
  }
  for i:=0; i < len(pal)/2; i += 1{
    if 'a' == pal[i] { continue}
    pal[i] = 'a'
    return string(pal)
  }
  pal[len(pal)-1] = 'b'
  return string(pal)
}
